src/Api/Controllers/OrganizationSponsorshipsController.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Api;
+using Bit.Core.Models.Api.Request;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Controllers
+{
+    [Route("organization/sponsorship")]
+    [Authorize("Application")]
+    public class OrganizationSponsorshipsController : Controller
+    {
+        private readonly IOrganizationSponsorshipService _organizationsSponsorshipService;
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationRepository _organizationRepository;
+        private readonly IOrganizationUserRepository _organizationUserRepository;
+        private readonly ICurrentContext _currentContext;
+        private readonly IUserService _userService;
+
+        public OrganizationSponsorshipsController(IOrganizationSponsorshipService organizationSponsorshipService,
+            IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+            IOrganizationRepository organizationRepository,
+            IOrganizationUserRepository organizationUserRepository,
+            IUserService userService,
+            ICurrentContext currentContext)
+        {
+            _organizationsSponsorshipService = organizationSponsorshipService;
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _organizationRepository = organizationRepository;
+            _organizationUserRepository = organizationUserRepository;
+            _userService = userService;
+            _currentContext = currentContext;
+        }
+
+        [HttpPost("{sponsoringOrgId}/families-for-enterprise")]
+        [SelfHosted(NotSelfHostedOnly = true)]
+        public async Task CreateSponsorship(Guid sponsoringOrgId, [FromBody] OrganizationSponsorshipRequestModel model)
+        {
+            var sponsoringOrg = await _organizationRepository.GetByIdAsync(sponsoringOrgId);
+
+            var sponsoringOrgUser = await _organizationUserRepository.GetByOrganizationAsync(sponsoringOrgId, _currentContext.UserId ?? default);
+
+            await _organizationsSponsorshipService.OfferSponsorshipAsync(sponsoringOrg, sponsoringOrgUser,
+                model.PlanSponsorshipType, model.SponsoredEmail, model.FriendlyName);
```

Comment:
The parameters of `OfferSponsorshipAsync` are pretty well named. Do we need to declare variables for all of these? This comment can be applied throughout most of this controller. 
```suggestion
            await _organizationsSponsorshipService.OfferSponsorshipAsync(
                await _organizationRepository.GetByIdAsync(sponsoringOrgId),
                await _organizationUserRepository.GetByOrganizationAsync(sponsoringOrgId, _currentContext.UserId ?? default),
                model.PlanSponsorshipType,
                model.SponsoredEmail,
                model.FriendlyName
            );
```

Reply:
While I am not a fan of single use variables everywhere I do think they add something to this one. It shows an order of operations that make it a bit more readable.

Reply:
Resolved offline -- Decided to go for it.